internal/source/git.go

```diff
@@ -209,11 +209,17 @@ func (f *billyFS) ReadFile(name string) ([]byte, error) {
 }
 
 func (f *billyFS) Open(path string) (fs.File, error) {
+	fi, err := f.Filesystem.Stat(path)
+	if err != nil {
+		return nil, err
+	}
+	if fi.IsDir() {
+		return &billyFile{nil, fi, err}, nil
+	}
```

Comment:
Not sure if you're still working, but I think we should:
- drop the `fiErr` field from the `billyFile` struct. It's unnecessary now because you're `err` ends up always being `nil` when a `billyFile` is returned.
- do something like `return &billyDir{billyFile{nil, fi}}` such that we return a `fs.File` from `Open` that also implements `fs.FileReadDir`, which seems to be the expectation of `fs.Open`.

Reply:
OK.  I'll try.

Reply:
I couldn't make it work without the `ReadDir`.   I might not do correctly.  Can I make a follow up issue and try later?

Reply:
I' doing one more try.

Reply:
It seems working now.  Thanks!